app/assets/javascripts/models/beta/iteration.js

```diff
@@ -182,13 +218,9 @@ export const groupBySprints = (stories = [], project, initialSprintNumber = 1, p
     const firstSprintIndex = 0;
     const isFromSprintInProgress = !Story.isUnstarted(story);
 
-    sprints = handleSprintsOverflow(
-      project,
-      sprints,
-      Story.getPoints(story),
-      initialSprintNumber,
-      velocity
-    );
+    if (hasOverflow(story, velocity) && !isFromSprintInProgress) {
+      return handleSprintsOverflow(project, sprints, story, velocity, initialSprintNumber);
+    }
 
     if (sprints.length === 0) {
       createFirstSprint(sprints, project, initialSprintNumber, velocity);
```

Comment:
What do you think about making `createFirstSprint` return the first sprint instead of mutating `sprints`? It could be something like this:

```js
if (sprints.length === 0) {
  sprints = [createFirstSprint(sprints, project, initialSprintNumber, velocity)];
}
```